dts/bindings/gpio/st,stm32mp2-gpio.yaml

```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2025 Savoir-faire Linux, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: STM32MP2 GPIO Controller
+
+compatible: "st,stm32mp2-gpio"
+
+include:
+  - name: gpio-controller.yaml
+    property-blocklist:
+      - ngpios
+  - name: base.yaml
+
+properties:
+  reg:
+    required: true
+
+  clocks:
+    description: |
+      The clocks property is optional for STM32MP2 GPIO controllers.
+      This property is not required when the Cortex-A core is responsible for
+      GPIO initialization and acts as the resource manager, as defined by the
+      RIF (Resource Isolation Framework) peripheral. Adding the GPIO clocks
+      in that case would only trigger an IAC event.
+      If the Cortex-M core is designated as the resource manager, then the
+      clocks property must be defined and added to the device tree and the
+      CONFIG_GPIO_HAS_CLOCK_CONTROL configuration option must be enabled.
```

Comment:
Ok for making `clocks` prop optional but no need to add this Kconfig option.
DTS bindings will ensure clocks is there when required, and DTS macros can be used to make the code depending on the prop presence.